docs/testing.rst

```diff
@@ -0,0 +1,140 @@
+Testing
+=======
+
+When building Flask apps that have integrated Flask-Dance you'll eventually run
+into the need to test your app. But if routes are guarded by the requirement of
+having a valid oauth2 session, i.e getting passed the `.authorized()` call on
+a provider, how do you test this?
+
+One possible way of solving this is by injecting the necessary information into
+the Flask session. The following examples assume you're using py.test and the
+Google provider, but the same trick applies to any of the supported providers.
+
+Creating a session
+------------------
+
+In your `conftest.py` create two fixtures, like this:
+
+.. code-block:: python
+
+    import time
+
+    import pytest
+
+    from your_app import create_app
+    from your_app.settings import Testing
+
+    fake_time = time.time()
+
+
+    @pytest.fixture
+    def app():
+        """Returns an app fixture with the testing configuration."""
+        app = create_app(config_object=Testing)
+        yield app
+
+
+    @pytest.fixture
+    def loggedin_app(app):
+        """Creates a logged-in test client instance."""
+        with app.test_client() as client:
+            with client.session_transaction() as sess:
+                sess['google_oauth_token'] = {
+                    'access_token': 'this is totally fake',
+                    'id_token': 'this is not a real token',
+                    'token_type': 'Bearer',
+                    'expires_in': '3600',
+                    'expires_at': fake_time + 3600,
+                }
+            yield client
+    
+This will inject the necessary information in order to ensure that Flask-Dance,
+requests-oauthlib and oauthlib2 believe there is a valid session, and one that
+won't need to be refreshed for another hour (3600 seconds). If your tests run
+longer than that, you'll need to adjust that value.
+
+The `fake_time` is created so that you can always refer to the same point in
+time throughout tests, and in any other fixture you might create. Alternatively
+you can use something like pytest-freezegun and then call any of the `time` and
```

Comment:
Please link to the pytest-freezegun website. Also, `time` and `datetime` need two sets of backticks.